File: BE/src/main/java/com/ssafy/e106/domain/subscription/dto/response/SubscriptionDetailResponse.java
package com.ssafy.e106.domain.subscription.dto.response;

import java.time.LocalDate;
import java.util.List;

import com.ssafy.e106.domain.subscription.entity.Subscription;

public record SubscriptionDetailResponse(
    Long subscriptionId,
    String subscriptionType,
    String bundleCode,
    String serviceName,
    String planName,
    String logoUrl,
    Integer monthlyPrice,
    Integer originalPrice,
    String billingCycle,
    LocalDate nextBillingDate,
    Integer daysUntilBilling,
    List<SubscriptionCoveredServiceResponse> coveredServices,
    List<SubscriptionRecentUsageResponse> recentUsage,
    SubscriptionRecommendationResponse recommendation,
    String cancelGuideUrl,
    String customerServicePhone,
    String contactEmail) {

  public static SubscriptionDetailResponse of(
      Subscription subscription,
      String serviceName,
      String planName,
      String logoUrl,
      Integer monthlyPrice,
      Integer originalPrice,
      String billingCycle,
      Integer daysUntilBilling,
      List<SubscriptionCoveredServiceResponse> coveredServices,
      List<SubscriptionRecentUsageResponse> recentUsage,
      SubscriptionRecommendationResponse recommendation) {
    return new SubscriptionDetailResponse(
        subscription.getSubscriptionId(),
        subscription.getSubscriptionType().name(),
        subscription.getBundle() == null ? null : subscription.getBundle().getCode(),
        serviceName,
        planName,
        logoUrl,
        monthlyPrice,
        originalPrice,
        billingCycle,
        subscription.getNextBillingDate(),
        daysUntilBilling,
        coveredServices,
        recentUsage,
        recommendation,
        subscription.getService() == null ? null : subscription.getService().getCancelGuideUrl(),
        subscription.getService() == null ? null : subscription.getService().getCustomerServicePhone(),
        subscription.getService() == null ? null : subscription.getService().getContactEmail());
  }
}

File: FE/app/src/main/java/com/ssafy/e106/data/dto/subscription/SubscriptionDetailResponse.kt
package com.ssafy.e106.data.dto.subscription

import kotlinx.serialization.Serializable

@Serializable
data class SubscriptionDetailResponse(
    val subscriptionId: Long,
    val subscriptionType: String = "SINGLE",
    val bundleCode: String? = null,
    val serviceName: String,
    val planName: String,
    val logoUrl: String? = null,
    val monthlyPrice: Int,
    val billingCycle: String? = null,
    val nextBillingDate: String? = null,
    val daysUntilBilling: Int? = null,
    val coveredServices: List<SubscriptionCoveredServiceResponse> = emptyList(),
    val recentUsage: List<SubscriptionRecentUsageResponse> = emptyList(),
    val recommendation: SubscriptionRecommendationResponse? = null,
    val cancelGuideUrl: String? = null,
    val customerServicePhone: String? = null,
    val contactEmail: String? = null,
)
